modules/mapbox-layers/src/deck-layer.js

```diff
@@ -23,29 +33,64 @@ export default class DeckLayer {
     };
   }
 
-  onAdd(map, gl) {
-    // console.log('onAdd', map, gl);
+  _getDeckInstance(gl) {
+    // Only create one deck instance per context
+    let deck = deckInstances.get(gl);
```

Comment:
This holds on to WebGL contexts and relies on `onRemove` to be rigorously implemented by mapbox (e.g. even when map is destroyed without first removing layers, which could easily not be the case). This is not great in dynamic map create/destroy scenarios.

Do we find it unappealing to store the data on the mapbox instance?